packages-exp/auth-exp/test/integration/flows/phone.test.ts

```diff
@@ -0,0 +1,182 @@
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { expect, use } from 'chai';
+import * as chaiAsPromised from 'chai-as-promised';
+
+import {
+  linkWithPhoneNumber,
+  PhoneAuthProvider,
+  reauthenticateWithPhoneNumber,
+  RecaptchaVerifier,
+  signInAnonymously,
+  signInWithPhoneNumber,
+  unlink,
+  updatePhoneNumber
+} from '@firebase/auth-exp/index.browser';
+import {
+  Auth,
+  OperationType,
+  ProviderId,
+  UserCredential
+} from '@firebase/auth-types-exp';
+import { FirebaseError } from '@firebase/util';
+
+import {
+  cleanUpTestInstance,
+  getTestInstance
+} from '../../helpers/integration/helpers';
+
+use(chaiAsPromised);
+
+// NOTE: These tests don't use a real phone number. In order to run these tests
+// you must whitelist the following phone numbers as "testing" numbers in the
+// auth console
+// https://console.firebase.google.com/u/0/project/_/authentication/providers
+//   • +1 (555) 555-1000, SMS code 123456
+//   • +1 (555) 555-2000, SMS code 654321
+
+const PHONE_A = {
+  number: '+15555551000',
+  code: '123456'
+};
+
+const PHONE_B = {
+  number: '+15555552000',
+  code: '654321'
+};
+
+describe('Integration test: phone auth', () => {
+  let auth: Auth;
+  let verifier: RecaptchaVerifier;
+  let fakeRecaptchaContainer: HTMLElement;
+
+  beforeEach(() => {
+    auth = getTestInstance();
+    fakeRecaptchaContainer = document.createElement('div');
+    document.body.appendChild(fakeRecaptchaContainer);
+    verifier = new RecaptchaVerifier(fakeRecaptchaContainer, undefined, auth);
+  });
+
+  afterEach(async () => {
+    await cleanUpTestInstance(auth);
+    document.body.removeChild(fakeRecaptchaContainer);
+  });
+
+  it('allows user to sign up', async () => {
+    const cr = await signInWithPhoneNumber(auth, PHONE_A.number, verifier);
+    const userCred = await cr.confirm(PHONE_A.code);
+
+    expect(auth.currentUser).to.eq(userCred.user);
+    expect(userCred.operationType).to.eq(OperationType.SIGN_IN);
+
+    const user = userCred.user;
+    expect(user.isAnonymous).to.be.false;
+    expect(user.uid).to.be.a('string');
+    expect(user.phoneNumber).to.eq(PHONE_A.number);
+  });
+
+  it('anonymous users can link (and unlink) phone number', async () => {
+    const { user } = await signInAnonymously(auth);
+    const { uid: anonId } = user;
+
+    const cr = await linkWithPhoneNumber(user, PHONE_A.number, verifier);
+    const linkResult = await cr.confirm(PHONE_A.code);
+    expect(linkResult.operationType).to.eq(OperationType.LINK);
+    expect(linkResult.user.uid).to.eq(user.uid);
```

Comment:
Please add an assertion to verify phoneNumber is correctly set, before unlinking it.

Reply:
Done